### PR TITLE
*: support the statement of "rename database" 

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve? 
Support the statement of "rename database".
`RENAME {DATABASE | SCHEMA} db_name TO new_db_name;`
The related MySQL [docs](http://download.nust.na/pub6/mysql/doc/refman/5.1/en/rename-database.html).

### What is changed and how it works?
Support the syntax of "rename database".

### Check List 

Tests 

 - Unit test

Code changes

 - Has interface methods change

### CODE REVIEW

File: ast/ddl.go [hunk 1]
```diff
@@ -194,6 +195,34 @@ func (n *DropDatabaseStmt) Accept(v Visitor) (Node, bool) {
 	return v.Leave(n)
 }
 
+// RenameDatabaseStmt is a statement to drop a database and all tables in the database.
```
Comment (djshow832) [inline ast/ddl.go]: REALLY????

Comment (zimulala) [reply ast/ddl.go]: @djshow832  Done.

File: ast/ddl.go [hunk 2]
```diff
@@ -194,6 +195,34 @@ func (n *DropDatabaseStmt) Accept(v Visitor) (Node, bool) {
 	return v.Leave(n)
 }
 
+// RenameDatabaseStmt is a statement to rename a database.
+// It supports in MySQL 5.1.7 and removed in MySQL 5.1.23.
+type RenameDatabaseStmt struct {
+	ddlNode
+
+	OldDB string
```
Comment (AilinKid) [inline ast/ddl.go:203]: s/string/CIStr

**Review by AilinKid — COMMENTED** [review]
Rest LGTM

File: ast/ddl.go (hunk 2 above)
Comment (zimulala) [reply ast/ddl.go:203]: Why do we need to do this?

**Comment by AilinKid** [comment]
> Rest LGTM

LGTM

**Comment by zimulala** [comment]
PTAL @djshow832

File: ast/ddl.go (hunk 2 above)
Comment (Deardrops) [reply ast/ddl.go:203]: @zimulala If not do this, we need convert it to `CIStr` when use this field in TiDB, e.g.
`model.NewCIstr(renameDatabaseStmt.OldDB)`.
If change to `CIStr` here, we can use it directly in TiDB.

Comment (zimulala) [reply ast/ddl.go:203]: I think it's OK. "create database" and "drop database" all use `string`.

**Review by Deardrops — APPROVED** [review]
LGTM

**Review by djshow832 — APPROVED** [review]
LGTM

**Pull request merged**